go/vt/discovery/endpointwatcher.go

```diff
@@ -0,0 +1,104 @@
+package discovery
+
+import (
+	"time"
+
+	log "github.com/golang/glog"
+	pbt "github.com/youtube/vitess/go/vt/proto/topodata"
+	"github.com/youtube/vitess/go/vt/topo"
+	"golang.org/x/net/context"
+)
+
+// NewEndPointWatcher returns an EndPointWatcher and starts refreshing.
+func NewEndPointWatcher(topoServer topo.Server, hc HealthCheck, cellsToWatch []string, refreshInterval time.Duration) *EndPointWatcher {
```

Comment:
This is a bad name, you're not watching endpoints, you're watching all the tablets in a cell, and producing endpoints. I'd suggest CellTabletsWatcher.

(the other watcher we talked about would be ShardReplicationWatcher).